add doc for record OOM alarm (#11577)
pingcap/docs-cn: system-variables.md

### `tidb_memory_usage_alarm_ratio`

- 作用域：GLOBAL
- 是否持久化到集群：是
- 默认值：`0.7`
- 范围：`[0.0, 1.0]`
- TiDB 内存使用占总内存的比例超过一定阈值时会报警，如果配置该选项为 `0` 或 `1`，则表示关闭内存阈值报警功能。
- 当内存阈值报警功能开启时，如果配置项 [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-从-v409-版本开始引入) 未设置，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * 系统内存大小`；如果 `server-memory-quota` 被设置且大于 0，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * server-memory-quota`。

> - 当 TiDB 检测到 tidb-server 的内存使用超过了阈值，则会认为存在内存溢出的风险，会将当前正在执行的所有 SQL 语句中内存使用最高的 10 条语句和运行时间最长的 10 条语句的相关信息, goroutine 栈信息以及 heap profile 记录到 TiDB 日志文件 [`filename`](tidb-configuration-file.md#filename) 所在目录中，并输出一条包含关键字 `tidb-server has the risk of OOM` 的日志。

Reviewer: 以及内存相关系统变量

Author: > 以及内存相关系统变量

done.

### `tidb_memory_usage_alarm_ratio`

- 作用域：GLOBAL
- 是否持久化到集群：是
- 默认值：`0.7`
- 范围：`[0.0, 1.0]`

> - 这个变量用于设置触发 tidb-server 内存告警的内存使用比率。默认情况下，当 TiDB 内存使用量超过总内存的 70% 且满足报警条件时，TiDB 会打印报警日志。

Reviewer 2: ```suggestion
- 这个变量用于设置触发 tidb-server 内存告警的内存使用比率。默认情况下，当 TiDB 内存使用量超过总内存的 70% 且满足[报警条件](/configure-memory-usage.md#tidb-server-内存占用过高时的报警)时，TiDB 会打印报警日志。
```

- 当配置该变量的值为 `0` 或 `1` 时，表示关闭内存阈值报警功能。
- 当配置该变量为 `0` 到 `1` 之间的值时，表示开启内存阈值报警功能：

    - 如果配置项 [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-从-v409-版本开始引入) 未设置，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * 系统内存大小`。
    - 如果配置项 `server-memory-quota` 被设置且大于 0，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * server-memory-quota`。

> ### `tidb_memory_usage_alarm_keep_record_num`

Reviewer 3: ```suggestion
### `tidb_memory_usage_alarm_keep_record_num` <span class="version-mark">从 v6.4.0 版本开始引入</span>
```

- 作用域：GLOBAL
- 是否持久化到集群：是
- 默认值：`5`
- 范围：`[1, 10000]`

> - 当 tidb-server 内存占用过高并触发报警时，TiDB 默认只保留最近 5 次报警时所生成的状态文件。你可以通过该变量调整该次数。

Reviewer 2: ```suggestion
- 当 tidb-server 内存占用超过内存报警阈值并触发报警时，TiDB 默认只保留最近 5 次报警时所生成的状态文件。你可以通过该变量调整该次数。
```

### `tidb_memory_usage_alarm_ratio`

- 作用域：GLOBAL
- 是否持久化到集群：是
- 默认值：`0.7`
- 范围：`[0.0, 1.0]`
- TiDB 内存使用占总内存的比例超过一定阈值时会报警，如果配置该选项为 `0` 或 `1`，则表示关闭内存阈值报警功能。
- 当内存阈值报警功能开启时：

    - 如果配置项 [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-从-v409-版本开始引入) 未设置，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * 系统内存大小`。
    - 如果配置项 `server-memory-quota` 被设置且大于 0，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * server-memory-quota`。

> - 当 TiDB 检测到 tidb-server 的内存使用超过了阈值，则会认为存在内存溢出的风险，会将当前正在执行的所有 SQL 语句中内存使用最高的 10 条语句和运行时间最长的 10 条语句的相关信息、goroutine 栈信息以及 heap profile 记录到 TiDB 日志文件 [`filename`](tidb-configuration-file.md#filename) 所在目录中，并输出一条包含关键字 `tidb-server has the risk of OOM` 以及内存相关系统变量的日志。
> - 其中包含的内存相关系统变量有：[`tidb_mem_oom_action`](#tidb_mem_oom_action-span-classversion-mark-v610-span)、[`tidb_mem_quota_query`](#tidb_mem_quota_query)、[`tidb_server_memory_limit`](#tidb_server_memory_limit)、[`tidb_analyze_version`](#tidb_analyze_version-span-classversion-mark-v510-span) 和 [`tidb_enable_rate_limit_action`](#tidb_enable_rate_limit_action)。

Reviewer 3: ```suggestion
- 当 TiDB 检测到 tidb-server 的内存使用超过了阈值，则会认为存在内存溢出的风险，会将当前正在执行的所有 SQL 语句中内存使用最高的 10 条语句和运行时间最长的 10 条语句的相关信息、goroutine 栈信息以及 heap profile 记录到 TiDB 日志文件 [`filename`](tidb-configuration-file.md#filename) 所在目录中，并输出一条包含关键字 `tidb-server has the risk of OOM` 以及以下内存相关系统变量的日志：
    - [`tidb_mem_oom_action`](#tidb_mem_oom_action-span-classversion-mark-v610-span)
    - [`tidb_mem_quota_query`](#tidb_mem_quota_query)
    - [`tidb_server_memory_limit`](#tidb_server_memory_limit)
    - [`tidb_analyze_version`](#tidb_analyze_version-span-classversion-mark-v510-span) 
    - [`tidb_enable_rate_limit_action`](#tidb_enable_rate_limit_action)
```

### `tidb_memory_usage_alarm_ratio`

- 作用域：GLOBAL
- 是否持久化到集群：是
- 默认值：`0.7`
- 范围：`[0.0, 1.0]`

> - TiDB 内存使用占总内存的比例超过一定阈值时会报警。当配置该变量的值为 `0` 或 `1` 时，表示关闭内存阈值报警功能。

Reviewer 3: ```suggestion
- TiDB 内存使用占总内存的比例超过一定阈值时会报警。
- 当配置该变量的值为 `0` 或 `1` 时，表示关闭内存阈值报警功能。
```

### `tidb_memory_usage_alarm_ratio`

- 作用域：GLOBAL
- 是否持久化到集群：是
- 默认值：`0.7`
- 范围：`[0.0, 1.0]`
- TiDB 内存使用占总内存的比例超过一定阈值时会报警。
- 当配置该变量的值为 `0` 或 `1` 时，表示关闭内存阈值报警功能。
- 当配置该变量为 `0` 到 `1` 之间的值时，表示开启内存阈值报警功能：

    - 如果配置项 [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-从-v409-版本开始引入) 未设置，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * 系统内存大小`。
    - 如果配置项 `server-memory-quota` 被设置且大于 0，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * server-memory-quota`。

> - 当 TiDB 检测到 tidb-server 的内存使用超过了阈值，则会认为存在内存溢出的风险，会将当前正在执行的所有 SQL 语句中内存使用最高的 10 条语句和运行时间最长的 10 条语句的相关信息、goroutine 栈信息以及 heap profile 记录到 TiDB 日志文件 [`filename`](tidb-configuration-file.md#filename) 所在目录中，并输出一条包含关键字 `tidb-server has the risk of OOM` 以及以下内存相关系统变量的日志：

Reviewer 3: 感觉如果把这段信息挪到 【## tidb-server 内存占用过高时的报警】 里会更清楚，因为除了这里提到的超过阈值，还需要满足【## tidb-server 内存占用过高时的报警】里提到的三个条件之一才会记录报警日志，你觉得呢

Author: yes, it is more clear.

    - [`tidb_mem_oom_action`](#tidb_mem_oom_action-span-classversion-mark-v610-span)
    - [`tidb_mem_quota_query`](#tidb_mem_quota_query)
    - [`tidb_server_memory_limit`](#tidb_server_memory_limit)
    - [`tidb_analyze_version`](#tidb_analyze_version-span-classversion-mark-v510-span) 
    - [`tidb_enable_rate_limit_action`](#tidb_enable_rate_limit_action)

### `tidb_memory_usage_alarm_keep_record_num`

- 作用域：GLOBAL
- 是否持久化到集群：是
- 默认值：`5`
- 范围：`[1, 10000]`

> - 这个变量设置了内存阈值报警功能开启时，当触发报警时最多保留几次最近 dump 下来的相关信息。

Reviewer 3: ```suggestion
- 当 tidb-server 内存占用过高并触发报警时，TiDB 默认只保留最近 5 次报警时所生成的状态文件。你可以通过该变量调整 TiDB 保留最近几次报警时所生成的状态文件。
```

### `tidb_memory_usage_alarm_ratio`

- 作用域：GLOBAL
- 是否持久化到集群：是
- 默认值：`0.7`
- 范围：`[0.0, 1.0]`
- 这个变量用于设置触发 tidb-server 内存告警的内存使用比率。默认情况下，当 TiDB 内存使用量超过总内存的 70% 且满足[报警条件](/configure-memory-usage.md#tidb-server-内存占用过高时的报警)时，TiDB 会打印报警日志。
- 当配置该变量的值为 `0` 或 `1` 时，表示关闭内存阈值报警功能。
- 当配置该变量为 `0` 到 `1` 之间的值时，表示开启内存阈值报警功能：

>     - 如果配置项 [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-从-v409-版本开始引入) 未设置，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * 系统内存大小`。

Reviewer 3: https://github.com/pingcap/docs-cn/pull/11565 里废弃了 `server-memory-quota`，被系统变量 [`tidb_server_memory_limit`](/system-variables.md#tidb_server_memory_limit-从-v640-版本开始引入) 所取代。
@mengxin9014 请看文档这里要不要也改成 `tidb_server_memory_limit` 系统变量？

Author: > 

done.

### `tidb_memory_usage_alarm_ratio`

- 作用域：GLOBAL
- 是否持久化到集群：是
- 默认值：`0.7`
- 范围：`[0.0, 1.0]`
- 这个变量用于设置触发 tidb-server 内存告警的内存使用比率。默认情况下，当 TiDB 内存使用量超过总内存的 70% 且满足[报警条件](/configure-memory-usage.md#tidb-server-内存占用过高时的报警)时，TiDB 会打印报警日志。
- 当配置该变量的值为 `0` 或 `1` 时，表示关闭内存阈值报警功能。
- 当配置该变量为 `0` 到 `1` 之间的值时，表示开启内存阈值报警功能：

>     - 如果系统变量 [`tidb_server_memory_limit`](/system-variables.md#tidb_server_memory_limit-从-v640-版本开始引入) 等于 0，且配置项 [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-从-v409-版本开始引入) 未设置，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * 系统内存大小`。

Reviewer 3: ```suggestion
    - 如果系统变量 [`tidb_server_memory_limit`](#tidb_server_memory_limit-从-v640-版本开始引入) 等于 0，且配置项 [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-从-v409-版本开始引入) 未设置，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * 系统内存大小`。
```

>     - 如果系统变量 [`tidb_server_memory_limit`](/system-variables.md#tidb_server_memory_limit-从-v640-版本开始引入) 等于 0，且配置项 [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-从-v409-版本开始引入) 未设置，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * 系统内存大小`。
>     - 如果系统变量 `tidb_server_memory_limit` 被设置且大于 0，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * tidb_server_memory_limit`。
>     - 如果系统变量 `tidb_server_memory_limit` 被设置且等于 0，配置项 `server-memory-quota` 被设置且大于 0，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * server-memory-quota`。

Reviewer 3: ```suggestion
    - 如果系统变量 [`tidb_server_memory_limit`](/system-variables.md#tidb_server_memory_limit-从-v640-版本开始引入) 等于 0，且配置项 [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-从-v409-版本开始引入) 未设置，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * 系统内存大小`。
        - 如果系统变量 `tidb_server_memory_limit` 等于 0，且配置项 `server-memory-quota` 被设置为大于 0，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * server-memory-quota`。
    - 如果系统变量 `tidb_server_memory_limit` 被设置为大于 0，则内存报警阈值为 `tidb_memory-usage-alarm-ratio * tidb_server_memory_limit`。
```
